src/ros_filter.cpp

```diff
@@ -391,71 +391,71 @@ namespace RobotLocalization
              "Integration time is " << std::setprecision(20) << currentTime << "\n"
              << measurementQueue_.size() << " measurements in queue.\n");
 
-    // If we have any measurements in the queue, process them
-    if(measurementsReady_.timed_wait(lock,boost::posix_time::milliseconds(filter_.getSensorTimeout()*1000)))
+    // If we do not have measurements in the queue wait until it is time to publish
+    if(measurementQueue_.empty())
```

Comment:
I'm leaning to trust on @ayrton04 here, but there's a lot of code changed here (which I know is part of the PR), and I'm concerned with possible issues running with simulated (gazebo or rosbag play --clock) and/or non-simulated (robot) time.
Have you tested it in both cases? I'm asking because I could probably give it a try in simulation (both with gazebo and rosbag play --clock). I can record the input and output topics to check they're fine (publish rate and values).